app/src/lib/dispatcher/app-store.ts

```diff
@@ -1507,28 +1507,28 @@ export class AppStore {
   }
 
   /** This shouldn't be called directly. See `Dispatcher`. */
-  public _deleteBranch(repository: Repository, branch: Branch): Promise<void> {
-    return this.withAuthenticatingUser(
```

Comment:
But this logic gave us the user to use for authentication, including non-GitHub server. What's the motivation for changing that?